### PR TITLE
Set acquisition era and processing string correctly in assign.py

### PR DESCRIPTION
**Pull request by areinsvo**
Edited assign.py to correctly set the acquisition era and processing string values when assigning. For merge ACDCs, no value is given. For all other workflows, era and procstring should inherit from the original workflow.

See also https://github.com/CMSCompOps/WmAgentScripts/issues/150

### CODE REVIEW

File: assign.py [hunk 1]
```diff
@@ -206,9 +206,9 @@ def main():
         elif options.sites == "t2":
             sites = T2_SITES
         else:
-            sites = SI.sites_ready
+            sites = [ s for s in SI.sites_ready if  (s.startswith('T1_') or s.startswith('T2_'))]
     else: 
-         sites = SI.sites_ready
+        sites = [ s for s in SI.sites_ready if  (s.startswith('T1_') or s.startswith('T2_')]
```
Comment (vlimant) [inline assign.py]: you can use sites_T1 and sites_T2 from siteInfo

Comment (prozober) [inline assign.py]: I am trying to use the script, and the first error I got is a "SyntaxError: invalid syntax" in this line.

Comment (vlimant) [inline assign.py]: there's a missing ")" indeed

File: assign.py [hunk 2]
```diff
@@ -256,6 +263,10 @@ def main():
             era = options.era
         elif not taskchain:
             era = wf.info['AcquisitionEra']
+        #Set era and procstring to none for merge ACDCs inside a task chain
+        if schema["RequestType"] == "Resubmission") and wfInfo.info["PrepID"].startswith("task") and "Merge" in schema["InitialTaskPath"].split("/")[-1]:
```
Comment (prozober) [inline assign.py]: Another SyntaxError, there is an additional ")"

Comment (prozober) [inline assign.py]: Despite of the SyntaxError, I am getting this error "NameError: global name 'wfInfo' is not defined", "wfInfo" does not exits,  the variable is "wf".

**Comment by amaltaro** [comment]
`pylint assign.py` works like a charm :)

**Comment by prozober** [comment]
Despite the syntax errors, I tried to assign an ACDC and I got this error:  "400 Bad Request - Bad output dataset name, check the processed dataset. Acquisition era cannot be None".
I attach in a file the parameters I sent.
[parameters.txt](https://github.com/CMSCompOps/WmAgentScripts/files/308892/parameters.txt)

**Comment by areinsvo** [comment]
Sorry about that! I just committed a version with the syntax errors fixed. In my rush to submit the pull request last night I clearly did not test the script like I should have. I'll now try to figure out what is going on with the error Paola reported.

**Comment by vlimant** [comment]
The part 
    if era is not None:
        params["AcquisitionEra"] = era
    if procstring is not None:
        params["ProcessingString"] = procstring
is not compatible with reqMgrClient that expects those parameters no matter what, so set them to the provided value. If None, reqMgrClient will pop them out, but will complain if not provided

**Comment by areinsvo** [comment]
Good catch. I fixed it and committed the changes. Paola is creating a backfill workflow for us to use to test that it works for merge ACDCs.

**Comment by vlimant** [comment]
I can merge this later today if I needed

**Comment by areinsvo** [comment]
Paola said she used the script to successfully assign the merge ACDC from earlier, so the script _should_ be good to go, but we are still going to perform the test with the backfill workflow to double check. I'm not sure that will be completed today though, and Jen probably needs this script to start assigning ACDCs.

File: assign.py [hunk 3]
```diff
@@ -231,17 +227,24 @@ def main():
 
         #Check to see if the workflow is a task chain or an ACDC of a taskchain
         taskchain = (schema["RequestType"] == "TaskChain") or ((schema["RequestType"] == "Resubmission") and "task" in schema["InitialTaskPath"].split("/")[1])
+
+        #Dealing with era and proc string
         if taskchain:
             # Setting the Era and ProcStr values per Task
             for key, value in schema.items():
                 if type(value) is dict and key.startswith("Task"):
```
Comment (prozober) [inline assign.py]: Hi Alli.

This line is never reached by TaskChain ACDCs. How can we iterate between the tasks of an ACDC? This line works for taskchain, not for resubmission's. E.g.:

TaskChain: https://cmsweb.cern.ch/couchdb/_utils/document.html?reqmgr_workload_cache/pdmvserv_task_HIG-RunIISpring16DR80-01109__v1_T_160608_190013_7236
Resubmission: https://cmsweb.cern.ch/couchdb/_utils/document.html?reqmgr_workload_cache/prozober_ACDC_task_HIG-RunIISpring16DR80-01109__v1_T_160613_164828_2440

Despite this, the ACDCs containing merge task are correctly assigned, as we tested on Friday. 

Regards.

**Pull request merged**